tiny_cnn/layers/max_pooling_layer.h

```diff
@@ -45,9 +45,11 @@ class max_pooling_layer : public feedforward_layer<Activation> {
         in_(in_width, in_height, in_channels),
         out_(in_width / pooling_size, in_height / pooling_size, in_channels)
     {
-        if ((in_width % pooling_size) || (in_height % pooling_size))
+        if ((in_width % pooling_size) || (in_height % pooling_size)) {
             pooling_size_mismatch(in_width, in_height, pooling_size);
+        }
 
+        set_worker_count(CNN_TASK_SIZE);
```

Comment:
@nyanp thx! this solved part of the problem, however, network multi_out test fails:

network:[ FAILED ] multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.888224 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.887691 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(166): actual_out2[1] == in[0]==in[1] failed for: 0.888835 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.887691 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.887691 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.887691 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(163): actual_out1[0] == in[0]&&in[1] failed for: 0.85465 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(166): actual_out2[1] == in[0]==in[1] failed for: 0.837397 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(163): actual_out1[0] == in[0]&&in[1] failed for: 0.85465 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(166): actual_out2[1] == in[0]==in[1] failed for: 0.837397 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(163): actual_out1[0] == in[0]&&in[1] failed for: 0.85465 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(166): actual_out2[1] == in[0]==in[1] failed for: 0.837397 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.888224 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.888224 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.887691 == 1
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(163): actual_out1[0] == in[0]&&in[1] failed for: 0.85465 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(166): actual_out2[1] == in[0]==in[1] failed for: 0.837397 == true
multi_out : /home/eriba/software/tiny-cnn/test/test_network.h(165): actual_out2[0] == in[0]^in[1] failed for: 0.887691 == 1